server/routes/release/actions.tsx

```diff
@@ -186,13 +191,12 @@ export default defineRoute(async (req, ctx) => {
 					</div>
 				)}
 				{release && isrcProvider && (
-					<div class='action'>
-						<SpriteIcon name='disc' />
-						<p>
-							<MagicISRC release={release} targetMbid={releaseMbid!} />
-							: Submit ISRCs from <a href={isrcProvider.url}>{isrcProvider.name}</a> to MusicBrainz
-						</p>
-					</div>
+					<ISRCSubmission
+						release={release}
+						targetMbid={releaseMbid}
+						isrcProvider={isrcProvider}
+						recordingsCache={mbRecordings}
+					/>
```

Comment:
With your refactoring `isrcProvider` is no longer needed outside this component and doesn't have to be passed in. We can infer it from `release` inside the component now.
But over there we can pass it along to the now internal component `MagicISRC` and avoid computing it twice.